src/bokeh/io/notebook.py

```diff
@@ -498,6 +500,41 @@ def publish_display_data(data: dict[str, Any], metadata: dict[Any, Any] | None =
     from IPython.display import publish_display_data
     publish_display_data(data, metadata, transient=transient, **kwargs)
 
+def remote_jupyter_proxy_url(port: int):
+    """Callable to configure Bokeh's show method when a proxy must be
+    configured.    If port is None we're asking about the URL
+    for the origin header.
+
+    Taken from documentation here:
+
+       https://docs.bokeh.org/en/latest/docs/user_guide/output/jupyter.html#jupyterhub
+
+    and made an implicit override when BOKEH_JUPYTER_EXTERNAL_URL is defined in
+    a user's environment to the external hostname of the hub, e.g. https://our-hub.edu
+
+    Args:
+       port(int):
+           random port generated by bokeh to avoid re-using recently closed ports
+
+       Returns: f(port: int) -> str
+           a callable which given a random port generates an external url which
+           will return through the proxy to the current notebook server.
```

Comment:
```suggestion
       Returns:
           str: a callable which given a random port generates an external url which
                 will return through the proxy to the current notebook server.
```

Reply:
This comment was way off so I rewrote it to capture (a) it's just a string and (b) it's a URL which will traverse the JupyterHub proxy to get back to this notebook session.